solc/CommandLineParser.cpp

```diff
@@ -840,13 +840,20 @@ bool CommandLineParser::processArgs()
 	else
 		m_options.input.mode = InputMode::Compiler;
 
-	if (
-		m_args.count(g_strExperimentalViaIR) > 0 &&
-		m_options.input.mode != InputMode::Compiler &&
-		m_options.input.mode != InputMode::CompilerWithASTImport
-	)
+	map<string, set<InputMode>> validOptionInputModeCombinations = {
+		{g_strErrorRecovery, {InputMode::Compiler, InputMode::CompilerWithASTImport}},
+		{g_strExperimentalViaIR, {InputMode::Compiler, InputMode::CompilerWithASTImport}},
+	};
+	string invalidOptions;
```

Comment:
We have a helper called `joinOptionNames()`. You can store options in a vector and then use that function to join them when you print a message.
```suggestion
	vector<string> invalidOptionsForCurrentInputMode;
```